app/controllers/searches_controller.rb

```diff
@@ -0,0 +1,15 @@
+class SearchesController < ApplicationController
+
+  def new
+    @search_type = params[:type]
+    unless params[:query].empty? || params[:query] == " "
+      if params[:type] == 'students'
+        @students = Student.where('first_name ILIKE ? OR last_name ILIKE ?', "%#{params[:query]}%", "%#{params[:query]}%")
+      else
+        @labs = Lab.where('name ILIKE ? AND cohort_id = ?', "%#{params[:query]}%", params[:cohort][:id])
+      end
+    end
+    render 'new'
+  end
+
```

Comment:
Create a Search model that has attributes #type, #result, #query, etc. search instance knows its type and can execute student or lab based search accordingly. pass search instance or search result instance in to view.